### PR TITLE
Add CudfSeriesGroupBy to optimize dask_cudf groupby-mean

### PR DESCRIPTION
**Pull request by rjzamora**
Adds a `CudfSeriesGroupBy` class to dask_cudf.  This allows the optimizations from #6248 to be used for `CudfSeriesGroupBy.mean` (in addition to `CudfDataFrameGroupBy.aggregate`).

### CODE REVIEW

File: python/dask_cudf/dask_cudf/groupby.py [hunk 1]
```diff
@@ -50,6 +71,26 @@ def aggregate(self, arg, split_every=None, split_out=1):
...
+            self.obj,
+            self.index,
+            {self._slice: "mean"},
+            split_every=split_every,
+            split_out=split_out,
+            dropna=self.dropna,
+            sep=self.sep,
+            sort=self.sort,
+            as_index=self.as_index,
+        )[self._slice]
```
Comment (kkraus14) [inline python/dask_cudf/dask_cudf/groupby.py:91]: Can this be replicated on the `CudfDataFrameGroupBy` class?

Comment (rjzamora) [reply python/dask_cudf/dask_cudf/groupby.py:91]: Yeah - For `CudfDataFrameGroupBy.mean`, we can just pass something like `{c: "mean" for c in self.obj.columns if c not in self.index}`

**Review by kkraus14 — APPROVED** [review]

**Comment by rjzamora** [comment]
@isVoid - The last commit should have added the "fast path" for `SeriesGroupBy.agg`

**Comment by isVoid** [comment]
> @isVoid - The last commit should have added the "fast path" for `SeriesGroupBy.agg`

Verified under context of #7180

**Comment by kkraus14** [comment]
@gpucibot merge

**Pull request merged**